refactor(test): add unit tests for agent/internal/crypt

joremysh opened the pull request:
* add unit test for agent/internal/crypt
* fixed typo in readme

related to #241

nandor-magyar commented on agent/internal/crypt/pgp_test.go:
```diff
@@ -0,0 +1,170 @@
+`
+)
+
+//const passphrase = []byte(`the passphrase of the private key`) // Passphrase of the privKey
```
deadcode?

joremysh replied:
removed

nandor-magyar reviewed (changes requested):
Please make sure that lint is executed, there are lint errors.

robot9706 commented on agent/internal/crypt/pgp_test.go:
```diff
@@ -0,0 +1,170 @@
+
+//const passphrase = []byte(`the passphrase of the private key`) // Passphrase of the privKey
+
+func TestDecryptSecrets(t *testing.T) {
```
In my opinion it would be better to separate the cases into multiple tests.

joremysh replied:
split the good case and fail case to 2 functions

polaroi8d replied:
@joremysh I think you have to split every scenario into different test cases. Collecting good test cases in 1 function as you mentioned is not the best solution.

robot9706 replied:
There's only one case in each test so it looks good to me 👍

polaroi8d commented:
@joremysh could you please sign your commits?

joremysh commented:
> @joremysh could you please sign your commits?

finally, it woks now

polaroi8d reviewed (changes requested):
Please split up your functions. One test case for 1 function. Otherwise good job, thanks! ✋🏻

joremysh commented:
> Please split up your functions. One test case for 1 function. Otherwise good job, thanks! ✋🏻

@polaroi8d I think there is only 1 test case in each function now.
Which lines should I split?

robot9706 reviewed (approved):
LGTM 🚀

polaroi8d commented on agent/internal/crypt/pgp_test.go:
```diff
@@ -0,0 +1,177 @@
+-----END PGP PUBLIC KEY BLOCK-----
+`
+	privKey = `-----BEGIN PGP PRIVATE KEY BLOCK-----
+Version: OpenPGP.js v4.10.10
```
Remove the Version and Comment section.

joremysh replied:
removed

polaroi8d commented:
> > Please split up your functions. One test case for 1 function. Otherwise good job, thanks! ✋🏻
> 
> @polaroi8d I think there is only 1 test case in each function now. Which lines should I split?

It's okay, just a tiny addition. Could you please remove the `Version` and `Comment` sections from the keys, following this PR? https://github.com/dyrector-io/dyrectorio/pull/295/files

joremysh commented:
> > > Please split up your functions. One test case for 1 function. Otherwise good job, thanks! ✋🏻
> > 
> > 
> > @polaroi8d I think there is only 1 test case in each function now. Which lines should I split?
> 
> It's okay, just a tiny addition. Could you please remove the `Version` and `Comment` sections from the keys, following this PR? https://github.com/dyrector-io/dyrectorio/pull/295/files

Sure, removed.

polaroi8d commented:
Please rebase and I'll merge. Thanks ✋🏻

polaroi8d commented:
Good job @joremysh thanks 😀 🤝
